src/main/java/javaslang/collection/ScanImpl.java

```diff
@@ -0,0 +1,49 @@
+package javaslang.collection;
+
+import java.util.Objects;
+import java.util.function.BiFunction;
+
+class ScanImpl {
```

Comment:
> possible, but we still need to have that reverse (actually should be reversed if we copy Scala) method. in an interface it will be visible.

In this case I created auxiliary classes, in this case

```
public interface Traversable<T> { ... }

interface TraversableModule<T> {
   
    class ScanUtil {
        static ... reverse(...) { ... }
    }
}
```